### PR TITLE
fix service list can not search by groupName only

### LINKED ISSUE #4282: 1.4.0 Service list Cannot query by service name
**Describe the bug**
image
image

### PR DESCRIPTION
**Pull request by mark4z**
Fixes #4282 

@KomachiSion It's new PR here

### CODE REVIEW

**Review by KomachiSion — DISMISSED** [review]

**Review by horizonzy — DISMISSED** [review]

**Review by Maijh97 — APPROVED** [review]
.

**Review by Maijh97 — CHANGES_REQUESTED** [review]
Please resolve CI error.

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 1]
```diff
@@ -73,4 +86,21 @@ public static void checkServiceNameFormat(String combineServiceName) {
...
+     * <p>serviceName | groupName | result</p>
+     * <p>serviceA    | groupA    | groupA@@serviceA</p>
+     * <p>nil         | groupA    | groupA@@</p>
+     * <p>nil         | nil       | @@</p>
+     *
+     * @return 'groupName@@serviceName'
+     */
+    public static String getGroupedNameOptional(final String serviceName, final String groupName) {
+        final String resultGroupedName = groupName + Constants.SERVICE_INFO_SPLITER + serviceName;
+        return resultGroupedName.intern();
```
Comment (chuntaojun) [inline api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:104]: If it is not a fixed variable, do not do this manually to get it into the constant pool

**Review by chuntaojun — CHANGES_REQUESTED** [review]

**Comment by chuntaojun** [comment]
Please append some unit test cases to prove your changes

**Comment by mark4z** [comment]
> Please append some unit test cases to prove your changes

ok.i will

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java (hunk 1 above)
Comment (mark4z) [reply api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java]: Already update

**Comment by mark4z** [comment]
> Please append some unit test cases to prove your changes

I just add ut for utils, Controller ut maybe need refactor.

**Review by sanxun0325 — APPROVED** [review]
LGTM

**Review by Maijh97 — APPROVED** [review]

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**